File: create_eb_zip.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');
const archiver = require('archiver');

const zipPath = path.join(__dirname, 'medvastr-backend-eb.zip');
const backendDir = path.join(__dirname, 'backend');

if (fs.existsSync(zipPath)) {
  fs.unlinkSync(zipPath);
}

const output = fs.createWriteStream(zipPath);
const archive = archiver('zip', { zlib: { level: 9 } });

output.on('close', function () {
  console.log(`Successfully created ${zipPath} (${archive.pointer()} total bytes)`);
});

archive.on('error', function (err) {
  throw err;
});

archive.pipe(output);

// 1. Add application.jar
const jarPath = path.join(backendDir, 'target', 'backend-0.0.1-SNAPSHOT.jar');
archive.file(jarPath, { name: 'application.jar' });

// 2. Add Procfile
const procPath = path.join(backendDir, 'Procfile');
archive.file(procPath, { name: 'Procfile' });

// 3. Add .ebextensions directory recursively with forward slashes
const ebDir = path.join(backendDir, '.ebextensions');
if (fs.existsSync(ebDir)) {
  const files = fs.readdirSync(ebDir);
  files.forEach(file => {
    const fullP = path.join(ebDir, file);
    archive.file(fullP, { name: `.ebextensions/${file}` });
  });
}

archive.finalize();

File: create_eb_zip.py
import os
import zipfile

root_dir = r"c:\Users\Admin\Desktop\Final_Medvastr"
zip_path = os.path.join(root_dir, "medvastr-backend-eb.zip")
timestamped_zip = os.path.join(root_dir, "medvastr-deploy-20260907-1635.zip")
backend_dir = os.path.join(root_dir, "backend")

def create_bundle(target_path):
    if os.path.exists(target_path):
        os.remove(target_path)

    with zipfile.ZipFile(target_path, 'w', zipfile.ZIP_DEFLATED) as zf:
        jar_path = os.path.join(backend_dir, "target", "backend-0.0.1-SNAPSHOT.jar")
        zf.write(jar_path, "application.jar")
        
        proc_path = os.path.join(backend_dir, "Procfile")
        zf.write(proc_path, "Procfile")

        eb_dir = os.path.join(backend_dir, ".ebextensions")
        if os.path.exists(eb_dir):
            for root, _, files in os.walk(eb_dir):
                for file in files:
                    full_p = os.path.join(root, file)
                    rel_p = os.path.relpath(full_p, backend_dir)
                    arcname = rel_p.replace("\\", "/")
                    zf.write(full_p, arcname)

    size_mb = os.path.getsize(target_path) / (1024 * 1024)
    print(f"Successfully created {target_path} ({size_mb:.2f} MB)")

create_bundle(zip_path)
create_bundle(timestamped_zip)
